litestar/openapi/config.py

```diff
@@ -41,7 +41,7 @@ class OpenAPIConfig:
 
     create_examples: bool = field(default=False)
     """Generate examples using the pydantic-factories library."""
-    openapi_controller: type[OpenAPIController] = field(default_factory=lambda: OpenAPIController)
+    openapi_controller: type[OpenAPIController] | None = field(default=None)
```

Comment:
What's the reason for this change?

Reply:
Just so we can detect in `__post_init__` if openapi_controller is explicitly set or not and avoid the path patch if it is

Reply:
I don't think this is the right way to go. 

Thoughts @litestar-org/maintainers

Reply:
I don't think so either.

With the controller typed as optional, everywhere that we reference the controller now needs to narrow away the `None` type union, e.g., `if self.openapi_config and self.openapi_config.openapi_controller:`. However, the controller isn't actually ever `None`, because it is always defined in `__post_init__`. So, yeh, doesn't feel quite right to me.

This is a convenience to save subclassing the openapi controller to define the path, right?

Reply:
Yeah that makes sense. I missed an easier check in `__post_init__`  with `if self.openapi_controller is OpenAPIController` so this change is gone now